src/WebJobs.Script.WebHost/Controllers/AdminController.cs

```diff
@@ -218,13 +218,19 @@ public override Task<HttpResponseMessage> ExecuteAsync(HttpControllerContext con
         [AllowAnonymous]
         public async Task<HttpResponseMessage> ExtensionWebHookHandler(string name, CancellationToken token)
         {
-            var provider = this._scriptHostManager.BindingWebHookProvider;
+            var provider = _scriptHostManager.BindingWebHookProvider;
 
             var handler = provider.GetHandlerOrNull(name);
             if (handler != null)
             {
-                var response = await handler.ConvertAsync(this.Request, token);
-                return response;
+                // key names for extension webhooks are named by convention
+                string keyName = $"{name}_extension".ToLowerInvariant();
+                if (!this.Request.HasAuthorizationLevel(AuthorizationLevel.System, keyName))
```

Comment:
This change requires ALL extension WebHooks to present a valid system key with the correct ID as a code query param or via headers